### ISSUE TITLE
[Feature] Failure classification

### ISSUE THREAD

**Issue by brunojbarros** [microsoft/playwright#27307]
### Use case
Hi everybody!

I'm thinking about what is the best way to re-use a test to perform a smoke test that includes:
1. Functional
2. Accessibility
3. """_Performance_""" (with many quotes)

The scope of the smoke would be to simply log in, navigate to each menu option, and verify that:
1. **load all components correctly**: I would check this via toHaveScreenshot
2. **no accessibility issues have been introduced**: I would check this via AxeBuilder.analyze()
3. **response times are still optimal**: I would verify this through testStep.duration and validating that the value is less than 3000 milliseconds. By the quotes I meant that, I'm not trying to do any kind of sophisticated performance testing, but simply to have a quick understanding that the performance was not abruptly degraded. I am closely following the k6 project with the idea of being able to implement it in the future.

What worries me is that, in any of the three assertions, the test would be marked as "Failed", something that makes sense as an assertion has not passed, but I do not consider the failure due to point 1 to be as important as that due to point 3.

I thought of some solutions, but I don't know if it is currently possible:
A- Add a tag after an assertion fails, then you could identify: bug-functional, bug-performance, bug-accessibility, etc.
B- Be able to create a new status as a test result, in addition to the existing ones (Passed / Failed / Flaky / Skipped), have a "Warning" status.

**Ideally:**
A new configuration should be added in `playwright.config.ts`, for example `errorTypes` (array of strings).

Then, you could configure:
`errorTypes: ['accessibility', 'functional', 'performance', 'ux/ui', 'etc'];`

Allow the test to be marked with multiple error types: `testInfo.errorTypes.push('accessibility');`

In the report, when selecting a "Failed", the error types that were configured are displayed.

![image](https://github.com/microsoft/playwright/assets/12484408/fb19ccb5-abc8-45ed-96cf-c0063b9582c2)

**Comment by brunojbarros** [comment]
> If this are not errors and the test should be passing you can try adding [attachments](https://playwright.dev/docs/api/class-testinfo#test-info-attachments) to your test with the warning details and then have your test report surface them. Alternatively, you can put tests whose failures are not considered critical into a separate directory/project and treat their failures differently. [Soft assertions](https://playwright.dev/docs/test-assertions#soft-assertions) may come in handy in this situation. Would that work?
> 
> The current set of test statuses is a fundamental part of Playwright Test and we are not planning to change them/make them extensible. For the scenarios like yours we recommend structuring your tests in accord with their severity.

Exact! I am using soft assertion in each case so that the test does not end and I can validate the three points indicated.

Dividing it into different projects or tests would imply repeating the same steps but only changing the assertion, for example, in one project or test, validating only accessibility and in another the "_performance_". Currently, given that the number of test cases is small, it would not be affected in terms of execution times, but repeating the same steps when there are 100 test cases, and also executing it in different browsers, will be a problem.

**Comment by yury-s** [comment]
Discussed this in a team meeting and decided that it's out of scope for Playwright, so closing this issue.
